timemachine/fe/free_energy.py

```diff
@@ -496,16 +496,20 @@ def _combine(self, ligand_values, host_values=None):
 def get_water_sampler_params(initial_state: InitialState) -> NDArray:
     """Given an initial state, return the parameters that define the nonbonded parameters of water with respect to the
     entire system.
-
-    Since we split the different components of the NB into ligand-water, ligand-protein, we want to use the ligand-water parameter
-    to ensure that the ligand component is correctly configured for the specific lambda window.
     """
-    summed_pot = get_bound_potential_by_type(initial_state.potentials, SummedPotential).potential
-    # TBD Figure out a better way of handling the jankyness of having to select the IxnGroup that defines the Ligand-Water
-    # Currently this just hardcodes to the first ixn group potential which is the ligand-water ixn group as returned by HostTopology.parameterize_nonbonded
-    ixn_group_idx = next(i for i, pot in enumerate(summed_pot.potentials) if isinstance(pot, NonbondedInteractionGroup))
-    assert isinstance(summed_pot.potentials[ixn_group_idx], NonbondedInteractionGroup)
-    water_params = summed_pot.params_init[ixn_group_idx]
+    nb_ixn_pot = get_bound_potential_by_type(initial_state.potentials, NonbondedInteractionGroup)
+    water_params = nb_ixn_pot.params
+
+    # If the protein is present, use the original protein parameters for the water sampler
+    if len(initial_state.protein_idxs):
+        prot_params = get_bound_potential_by_type(initial_state.potentials, Nonbonded).params[
+            initial_state.protein_idxs
+        ]
+        if isinstance(water_params, jax.Array):
```

Comment:
Do we need the if/else for jax gradients? If not could we just do `water_params = np.asarray(water_params)[protein_idxs] = ...`?